turbopack/crates/turbopack-core/src/ident.rs

```diff
@@ -3,13 +3,59 @@ use std::fmt::Write;
 use anyhow::Result;
 use once_cell::sync::Lazy;
 use regex::Regex;
+use serde::{Deserialize, Serialize};
 use turbo_rcstr::RcStr;
-use turbo_tasks::{ResolvedVc, TaskInput, ValueToString, Vc};
+use turbo_tasks::{NonLocalValue, ResolvedVc, TaskInput, ValueToString, Vc, trace::TraceRawVcs};
 use turbo_tasks_fs::FileSystemPath;
 use turbo_tasks_hash::{DeterministicHash, Xxh3Hash64Hasher, encode_hex, hash_xxh3_hash64};
 
 use crate::resolve::ModulePart;
 
+/// A layer identifies a distinct part of the module graph.
+#[derive(
+    Clone,
+    TaskInput,
+    Hash,
+    Debug,
+    DeterministicHash,
+    Eq,
+    PartialEq,
+    TraceRawVcs,
+    Serialize,
+    Deserialize,
+    NonLocalValue,
+)]
+pub struct Layer {
+    name: RcStr,
+    user_friendly_name: Option<RcStr>,
+}
+
+impl Layer {
+    pub fn new(name: RcStr) -> Self {
+        debug_assert!(!name.is_empty());
+        Self {
+            name,
+            user_friendly_name: None,
+        }
+    }
+    pub fn with_user_friendly_name(self, user_friendly_name: RcStr) -> Self {
```

Comment:
A builder seems overkill here? I would have just kept the `new_with_user_friendly_name(name, user_friendly_name)`

Reply:
removed